specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-03-01/storagecache.json

```diff
@@ -0,0 +1,1776 @@
+        },
+        "firmwareUpdateStatus": {
+          "description": "True if there is a firmware update ready to install on this Cache. The firmware will automatically be installed after firmwareUpdateDeadline if not triggered earlier via the upgrade operation.",
+          "enum": [
+            "available",
+            "unavailable"
+          ],
+          "x-ms-enum": {
+            "name": "FirmwareStatusType",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "type": "string"
+        },
+        "firmwareUpdateDeadline": {
+          "description": "Time at which the pending firmware update will automatically be installed on the Cache.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastFirmwareUpdate": {
+          "description": "Time of the last successful firmware update.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "pendingFirmwareVersion": {
+          "description": "When firmwareUpdateAvailable is true, this field holds the version string for the update.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "type": "object"
+    },
+    "UnknownProperties": {
+      "description": "Properties of an unknown type of Storage Target.",
+      "type": "object",
+      "additionalProperties": {
+        "type": "string"
+      }
+    },
+    "StorageTarget": {
+      "type": "object",
+      "description": "Type of the Storage Target.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/StorageTargetResource"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "x-ms-client-flatten": true,
+              "description": "StorageTarget properties",
+              "$ref": "#/definitions/StorageTargetProperties"
+            }
+          }
+        }
+      ]
+    },
+    "StorageTargetResource": {
+      "description": "Resource used by a Cache.",
+      "properties": {
+        "name": {
+          "description": "Name of the Storage Target.",
+          "readOnly": true,
+          "$ref": "#/definitions/ResourceName"
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID of the Storage Target."
+        },
+        "type": {
+          "description": "Type of the Storage Target; Microsoft.StorageCache/Cache/StorageTarget",
+          "readOnly": true,
+          "type": "string"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "StorageTargetProperties": {
+      "type": "object",
+      "description": "Properties of the Storage Target.",
+      "discriminator": "targetBaseType",
+      "properties": {
+        "targetBaseType": {
+          "description": "Type of the Storage Target.",
+          "$ref": "#/definitions/StorageTargetType"
+        },
```

Comment:
Consider setting it based on "kind". 
Kind is the top level property (similar to name, id, type)  supported for all resources by ARM.

Reply:
Since our RP is implemented polymorphic on property level instead of StorageTarget level. We actually tried at the beginning and it caused issues using SDKs generated by the swagger.